include/swift/AST/DiagnosticsSema.def

```diff
@@ -1433,6 +1433,8 @@ ERROR(invalid_ibinspectable,none,
       "only instance properties can be declared @%0", (StringRef))
 ERROR(invalid_ibaction_decl,none,
       "only instance methods can be declared @%0", (StringRef))
+ERROR(invalid_ibaction_decl_async,none,
+      "@%0 instance methods cannot be async", (StringRef))
```

Comment:
Nit: The other diagnostics all just say `@%0 methods`; let's do that here too.

```suggestion
      "@%0 methods cannot be async", (StringRef))
```

Reply:
Should probably also rename the diagnostic because the message isn’t hardcoded 🙃

Reply:
Done